iamclient/iamclient.go

```diff
@@ -302,6 +309,148 @@ func (client *Client) GetCertificate(
 	return response.GetCertUrl(), response.GetKeyUrl(), nil
 }
 
+// StartProvisioning starts provisioning.
+func (client *Client) StartProvisioning(nodeID, password string) (err error) {
+	log.WithField("nodeID", nodeID).Debug("Start provisioning")
+
+	var (
+		errorInfo *cloudprotocol.ErrorInfo
+		csrs      []cloudprotocol.IssueCertData
+	)
+
+	defer func() {
+		err = client.sender.SendStartProvisioningResponse(cloudprotocol.StartProvisioningResponse{
+			MessageType: cloudprotocol.StartProvisioningResponseMessageType,
+			NodeID:      nodeID,
+			ErrorInfo:   errorInfo,
+			CSRs:        csrs,
+		})
+		if err != nil {
+			err = aoserrors.Wrap(err)
+		}
+	}()
+
+	errorInfo = client.startProvisioning(nodeID, password)
+	if errorInfo != nil {
+		return aoserrors.New(errorInfo.Message)
+	}
+
+	csrs, errorInfo = client.createKeys(nodeID, password)
+	if errorInfo != nil {
+		return aoserrors.New(errorInfo.Message)
+	}
+
+	return nil
```

Comment:
return nil overrides err from defered function. I assume "return" without arguments should be used

Reply:
"return" without arguments is a crime. Don't do this.